### DISCUSSION TITLE
2-phynilpyridine is flat?

### DISCUSSION THREAD

**Discussion by MherMatevosyan** [rdkit/rdkit#4447] [Q&A]
Hi again, (sorry for many questions today)

I was working with PBF and found out something interesting for me (I am not a chemist by my training, so I might be missing something). Conformation generation for `2-phynilpyridine` gives a flat molecule (this was also the case for any `2-biphenyl-like compound` case), while I looked up at [Pubchem](https://pubchem.ncbi.nlm.nih.gov/compound/13887) and it seems to have some angle on the linking bond. I also exported pdb file to pymol and it seems completely flat. I would like to know if I am missing something on chemistry or Conformer Generation works incorrectly on this case.

Thanks in advance!

![image](https://user-images.githubusercontent.com/29042139/130965016-ab9f1be3-558b-452e-947d-06f22a5b2e3c.png)

**Comment by iwatobipen** [comment]
Hi MherMatevosyan,
I think EmbedMolecule will generates molecule with randomseed in default. So It will returns different conformers in each run.
If you would like to get same conformer, you should use randomSeed option.
https://rdkit.org/docs/source/rdkit.Chem.rdDistGeom.html
Or run some optimization method after generating conformer.
https://rdkit.org/docs/source/rdkit.Chem.rdForceFieldHelpers.html?highlight=mmffoptimizemoleculeconfs#rdkit.Chem.rdForceFieldHelpers.MMFFOptimizeMoleculeConfs
Thanks,
@iwatobipen

**Comment by manajit-das** [comment]
Here the linking bond is freely rotatable. If the ortho positions were substituted with sufficiently bulky group, it could have hinder the rotation and then it can't be flat. But for this molecule it can have a flat conformation. 
Hope it helps and still relevant to you.
Thanks

**Reply by MherMatevosyan** [reply]
Thanks for your answer. It is still relevant for me. The thing is that RDKit conformation seems to be always flat for the second molecule. As you can see in the picture below, I generated conformations 3 times and they all are still completely flat. I have tried to set `useRandomCoords=True` in `AllChem.EmbedMolecule` multiple times, and still every conformation is completely flat. I also tried to optimize with `AllChem.UFFOptimizeMolecule` and `AllChem.MMFFOptimizeMolecule`  after embedding and got the same results.
Is this ok from chemistry perspective? If yes, why is this the case?

![flat molecule](https://user-images.githubusercontent.com/29042139/211527407-6b989852-6532-4149-b6bf-35ade46fb2c2.png)

**Reply by manajit-das** [reply]
Answer: Yes,  this is okay from chemistry perspective.

The planarity in such molecule generally is determined by two competing factors (I) The resonance between the ring will try to stabilize the planar structure whereas (II) the steric hindrance among the ortho substituents will try to destabilize it and will make it non planar.

First of all there are slightly difference between the two molecules.
In case of **3-phynilpyridine** all the ortho positions are occupied with H atoms i.e. (four Hydrogen), which gives it some rotational barrier and the two rings are not exactly planar. 

Whereas in case of **2-phynilpyridine**, one of the ortho position is without any H atom (because it is N, not C) i.e. overall three Hydrogen atom. This makes the rotational barrier negligible, and the resonance factor overrides the steric factor.  
Though it is not a chemistry discussion platform, I tried to elaborate. You can read it in any standard  B.Sc./M.Sc. Chemistry textbook.
